zmq/log/handlers.py

```diff
@@ -80,9 +81,52 @@ def __init__(self, interface_or_socket, context=None):
             self.socket = self.ctx.socket(zmq.PUB)
             self.socket.bind(interface_or_socket)
 
+    @property
+    def root_topic(self):
+        return self._root_topic
+
+    @root_topic.setter
+    def root_topic(self, value):
+        self.setRootTopic(value)
+
+    def setRootTopic(self, root_topic):
+        """Set the root topic for this handler.
+
+        This value is prepended to all messages published by this handler, and it
+        defaults to the string 'root'. When you subscribe to this socket, you must
+        set your subscription to an empty string, or to at least the first letter of
+        the binary representation of this string to ensure you receive any messages
+        from this handler.
+
+        You may also set the root topic to an empty string. In this case, messages
+        will begin with the binary representation of the log level string (INFO, WARN, etc.).
+        Note that ZMQ SUB sockets can have multiple subscriptions.
+        """
+        self._root_topic = root_topic
+
+    @property
+    def formatters(self):
```

Comment:
Is there a reason formatters is a property? The getters and setters don't appear to do anything that a plain formatters attribute wouldn't do.

Reply:
Sure. My intent was to move formatters from a class-level attribute to a _protected attribute without breaking any existing code that expected them to be an attribute. Similar to what I did with root_topic to _root_topic. 

Really, I was hoping we could deprecate attribute access entirely and focus on the new setFormatter method instead. Right now, the only way to know how to set that attribute to the correct value is to look at the code inside handlers.py, learn it's a mapping indexed by the log level constant value, and with values of type logging.Formatter. I guess I'm proposing doing away with that direct attribute access. Maybe raise a DeprecationWarning on these properties? Or take the leap that no one is accessing them today, and just remove the properties. 

Either way is fine with me - if you'd prefer I revert formatters back to an attribute and remove these properties, I'll gladly make that happen. LMK.

Reply:
The nice thing about properties and attributes is that there's no distinction for API consumers. Making them attributes if there are no setter actions works, then if we discover a need for a setter action, making it a property later is completely transparent to the user. Let's leave formatters as an attribute.

Reply:
Gladly - should be done now.